### ISSUE TITLE
more information

### ISSUE THREAD

**Issue by maxfield-allison** [Dalayeth/NetworkFix#1]
Good morning! I'm assuming that this is a modification for steamsocket and sendqueuesize? Instead of needing to manually edit the valheim_assembly.dll every time there's an update. Can you fill me in more on how this functions?

**Comment by LongParsnip** [comment]
Have a look at the file "Network.cs"

This is using Harmony transpiler to patch ZDOMan.SendZDOs and ZSteamSocket.RegisterGlobalCallbacks, basically looks like a find and replace for the literals in valheim_assembly.dll to me.

**Comment by Dalayeth** [comment]
That is correct LongParsnip. This mod uses a transpiler patch to find where SendZDOs function loads its hardcoded value of 10240. It then replaces that value with the one in the config file, by default 30720. This is discuss in this [guide](https://jamesachambers.com/revisiting-fixing-valheim-lag-modifying-send-receive-limits/). In [this](https://jamesachambers.com/revisiting-fixing-valheim-lag-modifying-send-receive-limits/#comment-11709) comment chain patk88 also discusses that the best solution for him was to also change the games SendRateMin/SendRateMax which is using the same value. I added those options as configs as well. Let me know how it works for you as I have done limited testing as it's hard to get enough people together to stress it when I have time to develop.

**Comment by LongParsnip** [comment]
Hi Dalayeth,

Thanks for making the mod, it was a little annoying having to make the modification each time the game updated.
Patk88 in the comment section is me, I have your mod currently installed on my server and deployed to my clients and all is working as it was before.

For the config files I have changed the Min and Max rates back to 524288, I never found that changing the minimum had any measurable affect. But yeah it is difficult to test without 3 to 4 other people online (one thing I did find useful for testing is to increase the queue size in SendZDOs substantially which increases the network traffic).

**Comment by Dalayeth** [comment]
Prior to 1.1.0 it was just replacing a value and didn't need special code for the game and for the server but since 1.1.0 I have it set to check if it's a server or a client and patch accordingly. The mod should work for both clients and dedicated servers.

**Comment by Dalayeth** [comment]
Version 1.1.0 is out on [thunderstore](https://valheim.thunderstore.io/package/Dalayeth/NetworkFix/) right now.

**Comment by maxfield-allison** [comment]
You're the best
